daemon/gluster-blockd.c

```diff
@@ -43,6 +44,61 @@ static gbConfig *gbCfg;
 
 struct timeval TIMEOUT = {CLI_TIMEOUT_DEF, 0};  /* remote rpc call timeout, 5 mins */
 
+typedef struct globalData {
+  int    pfd;  /* pidfile file descriptor */
+  pid_t  chpid;
+} globalData;
+
+globalData ctx = {0,};
+
+
+void
+glusterBlockCleanGlobals(void)
+{
+  struct flock lock = {0, };
+
+
+  lock.l_type = F_UNLCK;
+  if (fcntl(ctx.pfd, F_SETLK, &lock) == -1) {
+    LOG("mgmt", GB_LOG_ERROR, "fcntl(UNLCK) on pidfile %s failed[%s]",
+        GB_LOCK_FILE, strerror(errno));
+  }
+
+  if (ctx.pfd) {
+    close(ctx.pfd);
+  }
+  glusterBlockDestroyConfig(gbCfg);
+}
+
+
+void
+onSigServerHandler(int signum)
+{
+  LOG("mgmt", GB_LOG_DEBUG,
+      "server process with (pid: %lu) received (signal: %s)",
+      getpid(), strsignal(signum));
+
+  svc_exit();
+
+  exit(EXIT_SUCCESS);
+}
+
+
+void
+onSigCliHandler(int signum)
+{
+  LOG("mgmt", GB_LOG_DEBUG,
+      "cli process with (pid: %lu) received (signal: %s)",
+      getpid(), strsignal(signum));
+
+  kill(ctx.chpid, signum);  /* Pass the signal to server process */
+
+  svc_exit();
+
+  return;
```

Comment:
And we could remove the  'return;' here.

Reply:
This is matter of style, to indicate we reach end of a void function.
Void functions can return without a value :-)

Reply:
Okay.